packages/repository/src/model.ts

```diff
@@ -137,6 +137,32 @@ export class ModelDefinition {
   }
 }
 
+function asJSON(value: any): any {
+  if (value == null) return value;
+  if (typeof value.toJSON === 'function') {
+    return value.toJSON();
+  }
+  // Handle arrays
+  if (Array.isArray(value)) {
+    return value.map(item => asJSON(item));
+  }
+  return value;
+}
+
+function asObject(value: any, options?: Options): any {
+  if (value == null) return value;
```

Comment:
Does `asObject` need check the flag for unknown properties as well like `toObject`?

`if (options && options.ignoreUnknownProperties === false) { // do the convertion}`

Reply:
No. The function is only used to handle 1st level properties.